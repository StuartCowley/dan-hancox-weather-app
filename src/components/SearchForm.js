import React from "react";
import PropTypes from "prop-types";
import "../styles/SearchForm.css";

function SearchForm({ searchText, setSearchText, onSubmit }) {
  const handleInputChange = (event) => setSearchText(event.target.value);

  return (
    <div className="search-form" onSubmit={onSubmit}>
      <input
        className="search-container"
        type="text"
        onChange={handleInputChange}
        value={searchText}
      />
      <button id="button" type="submit" onClick={onSubmit}>
        Search
      </button>
    </div>
  );
}

SearchForm.propTypes = {
  searchText: PropTypes.string.isRequired,
  setSearchText: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
};

export default SearchForm;
