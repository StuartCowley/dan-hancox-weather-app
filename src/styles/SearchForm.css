.search-form {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 10px 0;
}

.search-container {
  width: 80%;
  padding: 5px;
  font-size: 14px;
  border-radius: 10px;
  border: 1px solid lightgray;
  margin-bottom: 10px;
}

#button {
  width: 50%;
  padding: 5px;
  font-size: 14px;
  background-color: lightgray;
  color: rgb(0, 0, 0);
  border-radius: 10px;
  border: none;
  cursor: pointer;
  box-shadow: 2px 2px 5px gray;
}

#button:hover {
  background-color: gray;
}
