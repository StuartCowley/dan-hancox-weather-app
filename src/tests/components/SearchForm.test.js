import React from "react";
import { render } from "@testing-library/react";
import SearchForm from "../../components/SearchForm";

describe("it renders correctly", () => {
  const validProps = {
    searchText: "Bristol",
    setSearchText: () => {},
    onSubmit: () => {},
  };

  it("renders a search box and button", () => {
    const { asFragment } = render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.handleCitySearch}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
