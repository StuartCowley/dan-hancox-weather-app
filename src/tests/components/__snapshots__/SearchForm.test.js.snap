// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly renders a search box and button 1`] = `
<DocumentFragment>
  <div
    class="search-form"
  >
    <input
      class="search-container"
      type="text"
      value="Bristol"
    />
    <button
      id="button"
      type="submit"
    >
      Search
    </button>
  </div>
</DocumentFragment>
`;
